### PR TITLE
[💳] Unavailable cards in pledge screen

### PR DESCRIPTION
**Pull request by eoji**
# 📲 What
Disabling ineligible cards (based on project currency) in pledge screen.

# 🤔 Why
So users can't pledge with cards that aren't accepted.

# 🛠 How
- Added `ProjectUtils.acceptedCardType` to determine whether a card is accepted for a project's currency.
- Added `StoredCard.usdCardTypes` and `nonUsdCardTypes` but will be replaced by API.
- `RewardCardAdapter` now also takes in a project so the cards can be disabled if they're not accepted.
- Added `button_select.xml` for a disablable soft black button.
- Updated card expiration copy.
- Manually added `Not available` string because I am impatient.
- Tests.

# 👀 See
| USD Project 💲 | CAD Project 🇨🇦 |
| --- | --- |
| screenshot-2019-08-05_114828 | screenshot-2019-08-05_113355 |

# 📋 QA
**_Note:_** The v1 API does not return a project's localized country name so I'm just using `project.location.displayableName` because I can only work with what I have.

🇺🇸  All cards should be available for projects that are in USD.
🚫🇺🇸 Non USD projects only allow AMEX™, Mastercard™, and Visa™

# Story 📖
[Trello Story](https://trello.com/c/DnTtb7um/1482-ineligible-cards-on-pledge-screen)

### CODE REVIEW

**Comment by eoji** [comment]
[Ineligible Cards on Pledge Screen](https://trello.com/c/DnTtb7um/1482-ineligible-cards-on-pledge-screen)

File: app/src/main/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModel.kt [hunk 1]
```diff
@@ -1,13 +1,78 @@
...
+
+        private val buttonCTA = BehaviorSubject.create<Int>()
+        private val buttonEnabled = BehaviorSubject.create<Boolean>()
+        private val notAvailableCopyIsVisible = BehaviorSubject.create<Boolean>()
+        private val projectCountry = BehaviorSubject.create<String>()
+
+        init {
+            val project = this.cardAndProject
+                    .map { it.second }
+
```
Comment (Scollaco) [inline app/src/main/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModel.kt]: Extra line here

File: app/src/test/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModelTest.kt [hunk 2]
```diff
@@ -12,18 +14,68 @@ class RewardCardViewHolderViewModelTest : KSRobolectricTestCase() {
...
         this.vm.outputs.expirationDate().subscribe(this.expirationDate)
         this.vm.outputs.id().subscribe(this.id)
         this.vm.outputs.issuerImage().subscribe(this.issuerImage)
         this.vm.outputs.lastFour().subscribe(this.lastFour)
+        this.vm.outputs.notAvailableCopyIsVisible().subscribe(this.notAvailableCopyIsVisible)
+        this.vm.outputs.projectCountry().subscribe(this.projectCountry)
+    }
+
+    @Test
+    fun testButtonCTA() {
```
Comment (Scollaco) [inline app/src/test/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModelTest.kt]: Here a better name could be `testButtonCTATitle_whenCardIsAccepted`

File: app/src/test/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModelTest.kt [hunk 3]
```diff
@@ -12,18 +14,68 @@ class RewardCardViewHolderViewModelTest : KSRobolectricTestCase() {
...
+        setUpEnvironment(environment())
+        val creditCard = StoredCardFactory.discoverCard()
+
+        this.vm.inputs.configureWith(Pair(creditCard, ProjectFactory.project()))
+
+        this.buttonCTA.assertValue(R.string.Select)
+    }
+
+    @Test
+    fun testButtonCTA_whenCardNotAccepted() {
```
Comment (Scollaco) [inline app/src/test/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModelTest.kt]: Same here `testButtonCTATitle_when...`

File: app/src/test/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModelTest.kt [hunk 4]
```diff
@@ -67,9 +119,53 @@ class RewardCardViewHolderViewModelTest : KSRobolectricTestCase() {
...
         val creditCard = StoredCardFactory.discoverCard()
 
-        this.vm.inputs.configureWith(creditCard)
+        this.vm.inputs.configureWith(Pair(creditCard, ProjectFactory.project()))
 
         this.lastFour.assertValue("1234")
     }
 
+    @Test
+    fun testNotAvailableCopyIsVisible() {
```
Comment (Scollaco) [inline app/src/test/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModelTest.kt]: Here I would rename to `testNotAvailableCopyIsVisible_isFalse_whenCardIsAccepted`

File: app/src/test/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModelTest.kt [hunk 5]
```diff
@@ -67,9 +119,53 @@ class RewardCardViewHolderViewModelTest : KSRobolectricTestCase() {
...
+
+        val creditCard = StoredCardFactory.discoverCard()
+
+        this.vm.inputs.configureWith(Pair(creditCard, ProjectFactory.project()))
+
+        this.notAvailableCopyIsVisible.assertValue(false)
+    }
+
+    @Test
+    fun testNotAvailableCopyIsVisible_whenCardIsNotAccepted() {
```
Comment (Scollaco) [inline app/src/test/java/com/kickstarter/viewmodels/RewardCardViewHolderViewModelTest.kt]: And here `testNotAvailableCopyIsVisible_isTrue_when...`

**Pull request merged**